### PR TITLE
Copy over shape information into NeXus file when component is duplicated

### LINKED ISSUE #393: Duplicate button does not copy geometry information over to new component
Clicking "duplicate" doesn't seem to copy over geometry information in the nexus file or the tree view.

### PR DESCRIPTION
**Pull request by rerpha**
### Issue

Closes #393 

### Description of work

Copies over shape information into nexus file when the duplicate button is pressed. 

### Acceptance Criteria 

duplicate a component with some shape information and check that the new component has the shape information in the nexus file. Try this with both types of geometry. 

### UI tests

None added, no UI changes that aren't already covered by unit tests. 

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

**Comment by SkyToGround** [comment]
@rerpha I am on vacation so I am not able to immediately review the pull-request.

**Comment by matthew-d-jones** [comment]
@rerpha I've marked this as "in progress" until you've looked into the alternative approach to implementing this which we talked about yesterday.

File: nexus_constructor/component.py [hunk 1]
```diff
@@ -301,3 +301,13 @@ def get_shape(self) -> Optional[Union[OFFGeometry, CylindricalGeometry]]:
     def _remove_shape(self):
         if SHAPE_GROUP_NAME in self.group:
             del self.group[SHAPE_GROUP_NAME]
+
+    def duplicate(
+        self, components_list
```
Comment (matthew-d-jones) [inline nexus_constructor/component.py]: Missing type hint for `components_list`

File: nexus_constructor/component.py [hunk 2]
```diff
@@ -301,3 +301,13 @@ def get_shape(self) -> Optional[Union[OFFGeometry, CylindricalGeometry]]:
...
+    ) -> "Component":
+        return Component(
+            self.file,
+            self.file.duplicate_nx_group(
+                self.group, components_list
+            ),
+        )
```
Comment (matthew-d-jones) [inline nexus_constructor/component.py]: Missing new line at end of file

File: nexus_constructor/instrument.py [hunk 3]
```diff
@@ -72,6 +72,16 @@ def add_component(self, name: str, nx_class: str, description: str) -> Component
         component.description = description
         return component
 
+    def duplicate_component(
```
Comment (matthew-d-jones) [inline nexus_constructor/instrument.py]: This is not being used any more

File: nexus_constructor/nexus/nexus_wrapper.py [hunk 4]
```diff
@@ -141,6 +143,20 @@ def create_nx_group(
         self._emit_file()
         return group
 
+    @staticmethod
+    def duplicate_nx_group(
+        group_to_duplicate: h5py.Group, components_list
```
Comment (matthew-d-jones) [inline nexus_constructor/nexus/nexus_wrapper.py]: Missing type hint for `components_list`

File: nexus_constructor/nexus/nexus_wrapper.py [hunk 5]
```diff
@@ -141,6 +143,20 @@ def create_nx_group(
...
+    ) -> h5py.Group:
+        new_group_name = generate_unique_name(
```
Comment (matthew-d-jones) [inline nexus_constructor/nexus/nexus_wrapper.py]: I think this should be done in the duplicate method in component and the destination name passed as a input arg to this method. `duplicate_nx_group` may then be useful in other situations.

**Pull request merged**